### PR TITLE
dhcpcd: Fix patch url

### LINKED ISSUE #118253: Cannot download dhcpch patch
**Describe the bug**
An error occurs when trying to fetch dhcpcd patch
```
trying https://roy.marples.name/cgit/dhcpcd.git/patch/?id=114870290a8d3d696bc4049c32eef3eed03d6070
% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
Dload  Upload   Total   Spent    Left  Speed
0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
error: cannot download ?id=114870290a8d3d696bc4049c32eef3eed03d6070 from any mirror
error: build of '/nix/store/5n6flp80wx1dcmxkvpxii1l3c0nawqzd-?id=114870290a8d3d696bc4049c32eef3eed03d6070.drv' failed
```

**To Reproduce**
1. Fetch the patch in `nix repl`:
    ```
    pkgs = import <nixpkgs> {}
    pkgs.fetchpatch { 
        url = "https://roy.marples.name/cgit/dhcpcd.git/patch/?id=114870290a8d3d696bc4049c32eef3eed03d6070";
        sha256 = "0kzpwjh2gzvl5lvlnw6lis610p67nassk3apns68ga2pyxlky8qb"; 
    }
    ```
    Output:
    ```
    «derivation /nix/store/imrvq9jyhlfzqjnszywzyx00x0k30z1y-?id=114870290a8d3d696bc4049c32eef3eed03d6070.drv»
    ```
2. Try to realise the derivation (--check flag just in case the derivation is already built)
    ```
    nix-store --realise --check /nix/store/imrvq9jyhlfzqjnszywzyx00x0k30z1y-?id=114870290a8d3d696bc4049c32eef3eed03d6070.drv
    ```
    Output:
    ```
    checking outputs of '/nix/store/imrvq9jyhlfzqjnszywzyx00x0k30z1y-?id=114870290a8d3d696bc4049c32eef3eed03d6070.drv'...
    
    trying https://roy.marples.name/cgit/dhcpcd.git/patch/?id=114870290a8d3d696bc4049c32eef3eed03d6070
      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                     Dload  Upload   Total   Spent    Left  Speed
      0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
    curl: (22) The requested URL returned error: 404
    error: cannot download ?id=114870290a8d3d696bc4049c32eef3eed03d6070 from any mirror
    builder for '/nix/store/imrvq9jyhlfzqjnszywzyx00x0k30z1y-?id=114870290a8d3d696bc4049c32eef3eed03d6070.drv' failed with exit code 1
    error: build of '/nix/store/imrvq9jyhlfzqjnszywzyx00x0k30z1y-?id=114870290a8d3d696bc4049c32eef3eed03d6070.drv' failed
    ```

**Expected behavior**
The derivation can be built successfully.

**Screenshots**
N/A

**Additional context**
The package is needed on Linux and doesn't really affect MacOS.

**Notify maintainers**

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.6.0, macOS 10.15.7`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.4`
 - channels(user): `""`
 - channels(root): `"nixpkgs-20.09pre221814.10100a97c89"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by dshkyra**
###### Motivation for this change
Fix #118253. Update a link to dhcpch patch.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/dhcpcd/default.nix [hunk 1]
```diff
@@ -24,7 +24,7 @@ stdenv.mkDerivation rec {
 
   patches = [
     (fetchpatch {
-      url = "https://roy.marples.name/cgit/dhcpcd.git/patch/?id=114870290a8d3d696bc4049c32eef3eed03d6070";
+      url = "https://roy.marples.name/git/dhcpcd/patch/?id=114870290a8d3d696bc4049c32eef3eed03d6070";
       sha256 = "0kzpwjh2gzvl5lvlnw6lis610p67nassk3apns68ga2pyxlky8qb";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/dhcpcd/default.nix:28]: Did you check if the hash is still valid?

**Comment by SuperSandro2000** [comment]
Please sqash when merging.

File: pkgs/tools/networking/dhcpcd/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/tools/networking/dhcpcd/default.nix:28]:
```suggestion
      sha256 = "sha256-h1gg1QDi0pL0Qj+ld8Rb2trbsFmBI3XPxfX+/T5d0nA=";
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 118264` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>dhcpcd: <a href="https://termbin.com/ryl9">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/8ea4e964c54658778be846421ab0c706/raw/index.html">pretty log </a></li>
<li>pipework: log was empty</li>
  </ul>
</details>

**Comment by veprbl** [comment]
@GrahamcOfBorg eval

File: pkgs/tools/networking/dhcpcd/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/tools/networking/dhcpcd/default.nix:29]: @dshkyra the new hash is different and ofborg still uses the old.

**Pull request merged**